python/ray/data/random_access_dataset.py

```diff
@@ -37,8 +39,11 @@ def __init__(
         The constructor is a private API. Use ``dataset.to_random_access_dataset()``
         to construct a RandomAccessDataset.
         """
-        self._format = dataset.dataset_format()
-        if self._format not in ["arrow", "pandas"]:
+        ctx = DatasetContext.get_current()
+        assert not ctx.use_streaming_executor
```

Comment:
Should un-deprecate.

Reply:
We can call .cache() if needed to materialize the dataset prior to conversion to random access.

Reply:
Undeprecated. `cache` I don't think is necessary since we call `sort()` and then `get_internal_block_refs` below anyways.